Update url of the developer guide

bcl opened the pull request:
The developer guides have moved to a new url, fix all the references to the old one.

larskarlitski reviewed (approved):
Thanks!

(And sorry for not thinking about this one when doing the switch to main.)

brlane-rht committed:
Update url of the developer guide

The developer guides have moved to a new url, fix all the references to
the old one.

ochosi commented:
@bcl /  @ondrejbudai This change won't help much, because apart from `docs/on-premises/00-overview/index.md` these files are all pulled in periodically:

- https://github.com/osbuild/osbuild.github.io/blob/main/.github/workflows/pull-readmes.yml
- https://github.com/osbuild/osbuild.github.io/blob/main/readme-list

The real fix is to update the readmes in all respective repos.

ondrejbudai commented:
Chm, maybe we need a check in this repo when someone tries to update these files "manually". This is super-easy to miss.

ochosi commented:
Could add them to the `.gitignore` file and then ignore the `.gitignore` file in the pipeline? :innocent:

bcl commented:
Oh crap, I totally forgot about that.
